cpp/CMakeLists.txt

```diff
@@ -422,7 +424,7 @@ if(RAFT_COMPILE_NN_LIBRARY)
                INTERFACE_POSITION_INDEPENDENT_CODE ON
   )
 
-  target_link_libraries(raft_nn_lib PUBLIC faiss::faiss raft::raft)
+  target_link_libraries(raft_nn_lib PUBLIC faiss::faiss raft::raft raft_distance)
```

Comment:
`raft_nn_lib` sources make use of distance headers such as cosine.cuh/euclidean.cuh where it encounters build failure with missing cutlass headers.

Reply:
If it is raft_nn_lib srcs that have this dependency, the `target_link_libraries` usage should be `PRIVATE`.  Secondly it should be a dependency on the cutlass target and not `raft_distance`.

This currently creates a hard dependency for downstream consumers of raft, which breaks requests like `find_package(raft COMPONENTS nn)` since the `raft_distance` target won't exist for those users.

Reply:
so can I make it `target_link_libraries(raft_nn_lib PUBLIC faiss::faiss raft::raft nvidia::cutlass::cutlass)` ?
and then will I need to modify `get_cutlass.cmake` to add `rapids_export_package` to `raft-nn-lib-exports`

Reply:
you would want `target_link_libraries(raft_nn_lib PUBLIC faiss::faiss raft::raft PRIVATE nvidia::cutlass::cutlass)` and no  other change would be required.

Reply:
Thanks @cjnolet  and @robertmaynard for quick help and fixes!